src/averageAges.js

```diff
@@ -55,7 +63,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => !onlyWithSon
+    ? child.mother && people.find(person => person.name === child.mother)
+    : child.mother && people.find(person => person.name === child.mother)
+      && child.sex === 'm');
+
+  const sumOfAge = children.reduce((sum, child) => {
+    const mother = people.find(
+      person => person.name === child.mother);
+
+    return sum + child.born - mother.born;
+  }, 0);
+
+  return sumOfAge / children.length;
+}
+
+function findAverageAge(list) {
```

Comment:
I don't think `list` is a proper name for a variable here. The name should describe what is stored inside. I would come up with sth like that:
```suggestion
function findAverageAge(people) {
```